src/utils.cpp

```diff
@@ -1077,15 +1077,49 @@ std::string str_join(const std::vector<std::string> &list,
   return str;
 }
 
-bool is_numeric(const std::string &s)
+std::optional<std::variant<int64_t, uint64_t>> get_int_from_str(
+    const std::string &s)
 {
-  std::size_t idx;
-  try {
-    std::stoll(s, &idx, 0);
-  } catch (...) {
-    return false;
+  if (s.size() == 0) {
+    return std::nullopt;
+  }
+
+  static const auto hex_regex = std::regex("0[xX][0-9a-fA-F]+");
+
+  if (std::regex_match(s, hex_regex)) {
```

Comment:
Not sure regex is necessarily faster than an exception. Is it sufficient to check for `0x` or `0X` prefix?

Reply:
> Is it sufficient to check for 0x or 0X prefix?

Yeah, good idea. I'll just do that.